backend/app/api/src/helpers/AuthenticatedStructures.ts

```diff
@@ -125,44 +127,151 @@ export class AuthenticatedStructures {
     }
 
     static async organization(organization: Organization): Promise<OrganizationStruct> {
-        const organizationPeriod = await organization.getPeriod()
+        return (await this.organizations([organization]))[0];
+    }
+
+    static async organizations(organizations: Organization[]): Promise<OrganizationStruct[]> {
+        if(organizations.length === 0) {
+            return [];
+        }
+
+        //#region get period ids / organizations map
+        const periodIdOrganizationsMap = new Map<string, Organization[]>();
+
+        for(const organization of organizations) {
+            const periodId = organization.periodId;
+            const array = periodIdOrganizationsMap.get(periodId);
+            if(array !== undefined) {
+                array.push(organization);
+            } else {
+                periodIdOrganizationsMap.set(periodId, [organization]);
+            }
+        }
+        //#endregion
+
+        //#region get registration period for each organization
+        const organizationsWithRegistrationPeriods: [Organization, OrganizationRegistrationPeriod][] = [];
+
+        for(const [periodId, organizations] of periodIdOrganizationsMap.entries()) {
+            const organizationMap = new Map(organizations.map(o => [o.id, o]));
+
+            const result = await OrganizationRegistrationPeriod.where({
+                periodId,
+                organizationId: {
+                    sign: 'IN',
+                    value: Array.from(organizationMap.keys())
+                }
+            });
+
+            for(const organizationRegistrationPeriod of result) {
+                const organizationId = organizationRegistrationPeriod.organizationId;
+                const organization = organizationMap.get(organizationId);
+                if(organization) {
+                    organizationsWithRegistrationPeriods.push([organization, organizationRegistrationPeriod]);
+                }
+            }
+        }
+        //#endregion
+
+        //#region check if private data access
```

Comment:
In plaats van de regions, lijkt mij een menselijkere beschrijving vaak wat meer op zijn plaats - vooral met het doel om de code sneller te begrijpen zonder alle lijnen af te moeten lopen. Bv. wat hier handig zou zijn is een verduidelijking dat er een opsplitsing is tussen organizaties waarbij je de private meta mag zien, en organisaties waar dat niet het geval is. En met een kleine verduidelijking waarom webshops daar iets mee te maken hebben (in dit geval omdat enkel beheerders de privatemeta kunnen zien en dat altijd samenhangt met webshops - users zonder beheerdersrechten die een organizatie opvragen krijgen geen webshops en geen privatemeta doorgestuurd)